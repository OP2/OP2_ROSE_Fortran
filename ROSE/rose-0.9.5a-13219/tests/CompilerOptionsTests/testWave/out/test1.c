#include <stdio.h>
#include <stdbool.h>
bool obfuscated_condition();

volatile int input;

int fib(int n) {
  int a = 1, b = 1;
  bool (*cond)(void);
  cond = obfuscated_condition;
  if (n == 0) return 1;
  while (n > 1 && cond() != false ) {
    --n;
    int result = a + b;
    a = b;
    a +=input;
    b = result;
    a -=input;
  }
  return b;
}

int main()
{
  int x = input;
  int prod = 1;
  while (x > 0) {prod *= x; --x;}
  printf("factorial(input) = %d\n", prod);
  prod = fib(input);
  printf("fib(input) = %d\n", prod);
  return 0;
};

bool obfuscated_condition()
{
  return true;
} ;

