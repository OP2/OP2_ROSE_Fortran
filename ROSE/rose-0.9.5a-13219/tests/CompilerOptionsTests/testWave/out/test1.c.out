PreprocessingInfo test1.c line 1 col 1 : #include <stdio.h>
PreprocessingInfo test1.c line 2 col 1 : #include <stdbool.h>
PreprocessingInfo test1.c line 3 col 1 : bool expanded: _Bool
PreprocessingInfo test1.c line 9 col 3 : bool expanded: _Bool
PreprocessingInfo test1.c line 12 col 29 : false expanded: 0
PreprocessingInfo test1.c line 34 col 1 : bool expanded: _Bool
PreprocessingInfo test1.c line 36 col 10 : true expanded: 1
